internal/api/anonymous.go

```diff
@@ -0,0 +1,60 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/supabase/auth/internal/metering"
+	"github.com/supabase/auth/internal/models"
+	"github.com/supabase/auth/internal/storage"
+)
+
+func (a *API) SignupAnonymously(w http.ResponseWriter, r *http.Request) error {
+	ctx := r.Context()
+	config := a.config
+	db := a.db.WithContext(ctx)
+	aud := a.requestAud(ctx, r)
+
+	if config.DisableSignup {
+		return forbiddenError("Signups not allowed for this instance")
+	}
+
+	params, err := retrieveSignupParams(r)
+	if err != nil {
+		return err
+	}
+	params.Aud = aud
+	params.Provider = "anonymous"
+
+	// always call this outside of a database transaction as this method
+	// can be computationally hard and block due to password hashing
```

Comment:
Does this mean people can send `password` to `signInAnonymously()`? Can we check that if there's no email or phone there's also no password?

Reply:
no they can't send a `password` to `signInAnonymously` - this was just carried over from the regular signup flow

since there is no password, there won't be any password hashing performed, will remove the comment